static/app/components/replays/canvasReplayerPlugin.tsx

```diff
@@ -0,0 +1,149 @@
+    event: eventWithTime
+  ): Promise<void> {
+    if (!canvasEventMap.has(event)) {
+      const status = {
+        isUnchanged: true,
+      };
+      if ('commands' in data) {
+        const commands = await Promise.all(
+          data.commands.map(async c => {
+            const args = await Promise.all(
+              (c.args as CanvasArg[]).map(deserializeCanvasArg(imageMap, null, status))
+            );
+            return {...c, args};
+          })
+        );
+        if (status.isUnchanged === false) {
+          canvasEventMap.set(event, {...data, commands});
+        }
+      } else {
+        const args = await Promise.all(
+          (data.args as CanvasArg[]).map(deserializeCanvasArg(imageMap, null, status))
+        );
+        if (status.isUnchanged === false) {
+          canvasEventMap.set(event, {...data, args});
+        }
+      }
+    }
+  }
+
+  /**
+   * Clone canvas node, change parent document of node to current document, and
+   * insert an image element to original node (i.e. canvas inside of iframe).
+   *
+   * The image element is saved to `containers` map, which will later get
+   * written to when replay is being played.
+   */
+  function cloneCanvas(id: number, node: HTMLCanvasElement) {
+    const cloneNode = node.cloneNode() as HTMLCanvasElement;
+    canvases.set(id, cloneNode);
+    const el = document.createElement('img');
+    node.appendChild(el);
+    document.adoptNode(cloneNode);
+    containers.set(id, el);
+    return cloneNode;
+  }
+
+  return {
+    /**
+     * When document is first built, we want to preload canvas events. After a
+     * `canvas` element is built (in rrweb), insert an image element which will
+     * be used to mirror the drawn canvas.
+     */
+    onBuild: async (node, {id}) => {
+      if (!node) {
+        return;
+      }
+
+      if (node.nodeType === 9) {
+        // on full snapshot... deserialize all canvas events
+        const promises: Promise<any>[] = [];
+        containers.clear();
+        for (const event of events) {
+          if (
+            event.type === EventType.IncrementalSnapshot &&
+            event.data.source === IncrementalSource.CanvasMutation
+          ) {
+            promises.push(deserializeAndPreloadCanvasEvents(event.data, event));
+          }
+        }
+
+        await Promise.all(promises);
+      }
+
+      if (node.nodeName === 'CANVAS' && node instanceof HTMLCanvasElement) {
+        // Add new image container that will be written to
+        const el = document.createElement('img');
+        node.appendChild(el);
+        containers.set(id, el);
+      }
+    },
+
+    /**
+     * Mutate canvas outside of iframe, then export the canvas as an image, and
+     * draw inside of the image el inside of replay canvas.
+     */
+    handler: async (
+      e: eventWithTime,
+      _isSync: boolean,
+      {replayer}: {replayer: Replayer}
+    ) => {
+      if (e.type === 3) {
+        if (e.data.source === 9) {
```

Comment:
```suggestion
      if (e.type === EventType.IncrementalSnapshot &&
        e.data.source === IncrementalSource.CanvasMutation) {
```
nit: keep consistent with onBuild